### PR TITLE
Accelerate path diff in HiveCopyEntityHelper by reusing FileStatus.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@ydai1124 can you review? Thanks!

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java [hunk 1]
```diff
@@ -52,14 +55,16 @@
   protected final FileSystem fileSystem;
   protected final Properties properties;
 
-  public Set<Path> getPaths() throws IOException {
+  public Map<Path, FileStatus> getPaths() throws IOException {
 
     PathFindingMethod pathFindingMethod = PathFindingMethod.valueOf(
         this.properties.getProperty(HIVE_LOCATION_LISTING_METHOD, DEFAULT_HIVE_LOCATION_LISTING_METHOD).toUpperCase());
 
+    Map<Path, FileStatus> result = Maps.newHashMap();
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java]: nit: you can use `BiMap` here.

**Comment by ydai1124** [comment]
Minor comment. LGTM.

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java (hunk 1 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java]: I am concerned with the performance hit that a `BiMap` might introduce without clear benefit. Discussed offline.

**Pull request merged**